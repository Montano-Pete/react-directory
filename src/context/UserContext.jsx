import { createContext, useState } from 'react';
import { getProfile } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getProfile();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

export { UserContext, UserProvider };
