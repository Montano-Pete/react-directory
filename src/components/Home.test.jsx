import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';
import { ProvideAuth } from '../context/ProvideAuth';

it('should render a Home page', () => {
  const { container } = render(
    <MemoryRouter>
      <ProvideAuth>
        <Home />
      </ProvideAuth>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
