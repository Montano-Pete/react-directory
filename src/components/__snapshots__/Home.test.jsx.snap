// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Home page 1`] = `
<div>
  <div>
    <h1>
      This is the Home page!
    </h1>
  </div>
</div>
`;
